tests/drivers/dac/dac_loopback/src/test_dac.c

```diff
@@ -184,6 +184,23 @@
 #define ADC_CHANNEL_ID 0
 #endif
 
+#elif defined(CONFIG_BOARD_XG24_DK2601B) || \
+	defined(CONFIG_BOARD_XG24_RB4187C)
+
+/* VDAC0CH0 output is internally available on IADC_INPUT_DAC0 */
+
+#define DAC_DEVICE_NODE		DT_NODELABEL(vdac0)
+#define DAC_RESOLUTION		12
+#define DAC_CHANNEL_ID		0
+
+#define ADC_DEVICE_NODE		DT_NODELABEL(adc0)
+#define ADC_RESOLUTION		12
+#define ADC_GAIN		ADC_GAIN_1
+#define ADC_REFERENCE		ADC_REF_VDD_1
+#define ADC_ACQUISITION_TIME	ADC_ACQ_TIME_DEFAULT
+#define ADC_CHANNEL_ID		0
+#define ADC_1ST_CHANNEL_INPUT	0x20	/* IADC_INPUT_DAC0 */
```

Comment:
Include the dt-bindings header for ADC inside the silabs-specific section to avoid hard-coding the value (see the `nrf_saadc.h` inclusion above for inspiration).